sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVParameters.scala

```diff
@@ -71,6 +71,8 @@ private[sql] case class CSVParameters(parameters: Map[String, String]) extends L
 
   val nullValue = parameters.getOrElse("nullValue", "")
 
+  val codec = parameters.getOrElse("compression", parameters.getOrElse("codec", null))
```

Comment:
the other thing is that i'd create short-form names for the common options, e.g. "gzip" should become GzipCodec. You'd need to look into what the commonly supported formats are and come up with their short names. We should also make sure this is case insensitive.